File: gPC Expansion/gpc_germ.m
function xi=gpc_germ(V, n_sample, D)
syschar = V{1};
xi = zeros(n_sample, D);

% written by: Rajnish
for ct=1:D
    if syschar(ct)=='P'
        xi(:, ct) = unifrnd(0,1,n_sample,1);
    elseif syschar(ct)=='H'
        xi(:, ct) = randn(n_sample,1);     
    end
end

File: gPC Expansion using Python/gpc_germ.py
import numpy as np
def gpc_germ(V, n_sample, D):
    syschar = V[0]
    xi=np.zeros([n_sample,D],int)

    for ct in range(0,D):
        if syschar[ct]== "P":
            xi[:,ct]=np.transpose(np.array([np.random.uniform(0,1,n_sample)]))
        elif syschar[ct]== "H":
            xi[:,ct]=np.transpose(np.array([np.random.normal(0,1,n_sample)]))

    return(xi)

